ci_framework/roles/switch_config/tasks/onyx_config.yml

```diff
@@ -0,0 +1,113 @@
+---
+- name: Collect switch facts
+  mellanox.onyx.onyx_facts:
+  register: onyx_facts
+  failed_when: false
+
+- name: Check if JSON API is supported in switch
+  ansible.builtin.set_fact:
+    json_api_support: |-
+      {%- if onyx_facts['rc'] == 1 and 'command \"json-print\"' in onyx_facts['module_stderr'] -%}
+        {{ False | bool }}
+      {%- else -%}
+        {{ True | bool }}
+      {%- endif -%}
+
+- name: Print JSON API support
+  ansible.builtin.debug:
+    var: json_api_support
+
+- name: Quit if JSON API is supported
+  ansible.builtin.fail:
+    msg: "This playbook was tested on non JSON API enabled switch and not using native onyx modules to perform actions"
+  when: json_api_support
+
+- name: MLNX-OS non JSON API enabled configuration block
+  become: true
+  when: not json_api_support
```

Comment:
What's the point of this when clause if we fail in the previous task?

Reply:
it is not actually necessary but I think it makes the code a bit more understandable. I can remove it if it is important in your opinion.

Reply:
No problem keeping it, is redundant IMHO, but is harmless.